### PR TITLE
V0.0.1

### PR DESCRIPTION
**Pull request by deeptanshu97rathore**

### CODE REVIEW

File: src/http/index.js [hunk 1]
```diff
@@ -60,7 +61,18 @@ class HttpCommunication {
       return finalURL.toString();
     }
 
-    async makeReqeust(params) {
+    populateHeadersFromContext(ctx) {
+      const customHeaders = {
+        'X-Q-TRACEID': ctx && ctx.traceId ? ctx.traceId : uuid(),
```
Comment (akshendra) [inline src/http/index.js:66]: ternary with conditional

File: src/http/index.js [hunk 2]
```diff
@@ -60,7 +61,18 @@ class HttpCommunication {
...
+      };
+      if (ctx) {
+        if (ctx.userId) headers['X-Q-USERID'] = ctx.userId;
```
Comment (akshendra) [inline src/http/index.js:69]: BUG: `headers` is not defined

**Review by akshendra — CHANGES_REQUESTED** [review]

File: src/http/index.js [hunk 3]
```diff
@@ -26,17 +27,28 @@ class HttpCommunication {
       this.contextStorage = contextStorage;
     }
 
+    static getRequestContext(req, customContextValue) {
+      return {
+        traceId: (req.headers && req.headers['X-Q-TRACEID']) ? req.headers['X-Q-TRACEID'] : uuid(),
+        userId: (req.user && req.user.userId) ? String(req.user.userId) : null,
+        ab: (req.headers && req.headers['Q-AB-ROUTE']) ? req.headers['Q-AB-ROUTE'] : null,
+        reqStartTime: process.hrtime(),
```
Comment (akshendra) [inline src/http/index.js:35]: its being deprecated https://nodejs.org/docs/latest-v16.x/api/process.html#processhrtimebigint

Comment (deeptanshu97rathore) [reply src/http/index.js:35]: Do you think we should use `hrtime.bigint();`

**Pull request closed without merging**